src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -230,10 +237,11 @@ void create_actor(SchedulingAlgorithmState *algorithm_state,
                   LocalSchedulerClient *worker) {
   /* This will be freed when the actor is removed in remove_actor. */
   LocalActorInfo *entry = (LocalActorInfo *) malloc(sizeof(LocalActorInfo));
+  memset(entry, 0, sizeof(LocalActorInfo));
```

Comment:
Yeah, this is a bit awkward, but it has to be this way for now while we're mixing C structs and C++ objects. When we `malloc` the `LocalActorInfo`, it might get initialized with garbage values. A `std::unique_ptr` that is a member of the struct will also get initialized with garbage values, and it will seem to hold a pointer, but the pointer isn't valid. When we reset the `std::unique_ptr` to point to the actual object we want it to point to, it tries to clean up the invalid pointer under the hood. So, I used `memset` so that it gets initialized to hold a NULL pointer.